packages/lesswrong/server/resolvers/userResolvers.ts

```diff
@@ -21,6 +21,82 @@ import { getAnalyticsConnection } from "../analytics/postgresConnection";
 import GraphQLJSON from 'graphql-type-json';
 import { rateLimitDateWhenUserNextAbleToComment, rateLimitDateWhenUserNextAbleToPost } from '../rateLimits/utils';
 import { RateLimitInfo } from '../rateLimits/types';
+import VotesRepo, { RecentVoteInfo } from '../repos/VotesRepo';
+import uniq from 'lodash/uniq';
+// import { groupBy } from 'underscore';
+
+export type RecentKarmaInfo = {
+  recentKarma: number, 
+  recentPostKarma: number,
+  recentCommentKarma: number,
+  downvoterCount: number, 
+  postDownvoterCount: number,
+  commentDownvoterCount: number,
+}
+
+async function getLatest20contentItems (votes: RecentVoteInfo[]): Promise<RecentVoteInfo[]> {
+  const sortedVotes = votes.sort((a, b) => b.postedAt.getTime() - a.postedAt.getTime())
+  const uniqueVoteDocumentIds = uniq(sortedVotes.map((vote) => vote.documentId))
+  const uniqueVoteDocumentsByIdSlice = uniqueVoteDocumentIds.slice(0, 20)
+  return sortedVotes.filter((vote) => uniqueVoteDocumentsByIdSlice.includes(vote.documentId))
+}
+
+
+export async function getRecentKarmaInfo (userId: string): Promise<RecentKarmaInfo> {
+  const votesRepo = new VotesRepo()
+  const allVotes = await votesRepo.getVotesOnRecentContent(userId)
+  const top20documentVotes = await getLatest20contentItems(allVotes)
+
+  const nonuserIDallVotes = allVotes.filter((vote: RecentVoteInfo) => vote.userId !== userId)
```

Comment:
Is this filtering out self-votes? If so, we should probably do it in the DB query, if we're always going to filter them out and not use them anywhere

Reply:
(the name on the variable isn't great :( )

Reply:
In the current implementation, if you remove the self-votes from the DB query, then you miss out on any comments where only the user self-upvoted them.

BUT, I do acknowledge that relying on their self-upvotes existing to fetch their last 20 comments is pretty brittle. I updated while writing this that we should probably just actually just do a different query, but I figured I would chat with you about what exact query to do instead. Seemed like it required a bit of thinking.